File: py/007.py
''' Problem 007:
    By listing the first six prime numbers: 2, 3, 5, 7, 11
    and 13, we can see that the 6th prime is 13. What is the
    10000th prime number? '''
    
def solution(n) -> int:
    n = int(n)
    # The function takes one input n, which represents the
    # nth prime number.

    # Skip the first three primes.
    prime = 5
    count = 3

    while (count < n):
        prime += 1
        if (isPrime(prime)):
            count += 1
    
    return prime

# Function to check for primes
def isPrime(n):    
    # The function takes one input n, which is
    # to be verified as a prime number.
    # Note that n > 3.

    # For primes 2 and 3
    if (n % 2 == 0 or n % 3 == 0):
        return False

    # For n >= 5:
    i = 5
    while i * i <= n:
        if (n % i == 0 or n % (i + 2) == 0):
            # n is composite
            return False
        
        i += 6

    # n is prime
    return True

def main():
    n = input("Type an integer: ")
    print(f"The {n}th prime is is {solution(n)}")
    
if __name__ == "__main__":
    main()
    
# Solved 20th July 2025.

File: cpp/007.cpp

/*  Problem 007:
    By listing the first six prime numbers: 2, 3, 5, 7, 11
    and 13, we can see that the 6th prime is 13. What is the
    10000th prime number? */
#include <iostream>
using namespace std;

using num = unsigned long long;

// Function to check for primes
inline bool isPrime(num n) {    
    // The function takes one input n, which is
    // to be verified as a prime number.
    // Note that n > 3.

    // For primes 2 and 3
    if (n % 2 == 0 || n % 3 == 0) {
        return false;
    
    }

    // For larger primes
    for (num i = 5; i * i <= n; i += 6) {
        if (n % i == 0 || n % (i + 2) == 0) {
            // n is composite
            return false;
        }
    }

    // n is prime
    return true;
}

num solution(num n) {
    // The function takes one input n, which represents the
    // nth prime number.

    // Skip the first three primes.
    num prime = 5;
    num count = 3;
    
    while (count < n) {
        prime++;
        if (isPrime(prime)) {
            count++;
        }
    }
    
    return prime;
}

int main(void) {
    num n;
    cout << "Input an integer: ";
    cin >> n;

    // Pardon the grammatical inaccuracy.
    cout << "The " << n << "th prime is " << solution(n) << endl;

    return 0;
}

// Solved 20th July 2025.